src/types/check_config.rs

```diff
@@ -82,11 +89,19 @@ impl CheckConfig {
     pub fn slots(&self) -> Vec<usize> {
         let interval_secs = self.interval as usize;
 
-        let first_slot = self
-            .subscription_id
-            .as_u128()
-            .checked_rem(self.interval as u128)
-            .unwrap() as usize;
+        // We deterministically produce a new uuid here based on the subscription id and use it to
+        // help distribute subscriptions to slots. Without this, we end up distributing all checks
+        // associated with a specific partition into the same slots.
+        //
+        // We have to use this instead of the subscription id itself because we already use the
+        // subscription id to partition checks.
+        let first_slot = Uuid::new_v5(
+            &SUBSCRIPTION_SLOT_NAMESPACE,
+            self.subscription_id.as_bytes(),
+        )
```

Comment:
This is the only change here, the rest of this pr is updating tests to choose appropriate uuids